pkg/frontend/openshiftcluster_putorpatch_test.go

```diff
@@ -3300,3 +3300,137 @@ func TestEnrichClusterSystemData(t *testing.T) {
+					TenantID: "11111111-1111-1111-1111-111111111111",
+				},
+			},
+		})
+		f.AddOpenShiftClusterDocuments(&api.OpenShiftClusterDocument{
+			Key: strings.ToLower(testdatabase.GetResourcePath(mockSubID, "resourceName")),
+			OpenShiftCluster: &api.OpenShiftCluster{
+				ID:       testdatabase.GetResourcePath(mockSubID, "resourceName"),
+				Name:     "resourceName",
+				Type:     "Microsoft.RedHatOpenShift/openShiftClusters",
+				Location: "eastus",
+				Tags:     map[string]string{"tag": "will-not-be-kept"},
+				Properties: api.OpenShiftClusterProperties{
+					ProvisioningState: api.ProvisioningStateSucceeded,
+					ClusterProfile: api.ClusterProfile{
+						Domain:               "example.aroapp.io",
+						ResourceGroupID:      fmt.Sprintf("/subscriptions/%s/resourcegroups/clusterResourceGroup", mockSubID),
+						FipsValidatedModules: api.FipsValidatedModulesDisabled,
+					},
+					MasterProfile: api.MasterProfile{
+						EncryptionAtHost: api.EncryptionAtHostDisabled,
+						VMSize:           api.VMSize("Standard_D32s_v3"),
+						SubnetID:         fmt.Sprintf("/subscriptions/%s/resourcegroups/network/providers/Microsoft.Network/virtualNetworks/test-vnet/subnets/master", mockSubID),
+					},
+					ServicePrincipalProfile: api.ServicePrincipalProfile{
+						ClientID:     "00000000-0000-0000-1111-000000000000",
+						ClientSecret: "00000000-0000-0000-0000-000000000000",
+					},
+					NetworkProfile: api.NetworkProfile{
+						PodCIDR:     "10.0.0.0/16",
+						ServiceCIDR: "10.1.0.0/16",
+					},
+					APIServerProfile: api.APIServerProfile{
+						Visibility: api.VisibilityPrivate,
+					},
+					IngressProfiles: []api.IngressProfile{
+						{
+							Visibility: api.VisibilityPublic,
+						},
+					},
+					OperatorFlags: api.OperatorFlags{},
+				},
+				SystemData: api.SystemData{
+					CreatedBy:          "ExampleUser",
+					CreatedByType:      api.CreatedByTypeApplication,
+					CreatedAt:          &createTime,
+					LastModifiedBy:     "ExampleUser",
+					LastModifiedByType: api.CreatedByTypeApplication,
+					LastModifiedAt:     &lastModifyTime,
+				},
+				Identity: &api.Identity{},
+			},
+		})
+	}
+
+	ctx := context.Background()
+
+	ti := newTestInfra(t).
+		WithOpenShiftClusters().
+		WithSubscriptions().
+		WithAsyncOperations()
+	defer ti.done()
+
+	err := ti.buildFixtures(fixture)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f, err := NewFrontend(ctx, ti.audit, ti.log, ti.env, ti.asyncOperationsDatabase, ti.clusterManagerDatabase, ti.openShiftClustersDatabase, ti.subscriptionsDatabase, ti.openShiftVersionsDatabase, api.APIs, &noop.Noop{}, &noop.Noop{}, nil, nil, nil, nil, ti.enricher)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, tt := range []struct {
+		name        string
+		identityURL []string
+		expected    *api.OpenShiftCluster
+		wantError   string
+	}{
+		{
+			name:        "pass - identity URL is nil",
+			identityURL: nil,
+			expected:    &api.OpenShiftCluster{},
+			wantError:   "",
+		},
+		// {
```

Comment:
This test fails because when `persistIdentityUrl` is called value for `Identity` in cluster object is nil. Thus it fails for `IdentityURL` assignment here:- `cluster.Identity.IdentityURL = identityURL[0]`

It would be a good idea to perform a MIWI cluster check on the basis of `PlatformWorkloadIdentityProfile` or `cluster.Identity`. And a test case for cluster object without Identity profile and the for already existing in this file should have the Identity profile in the cluster object like:-
```
cluster := &api.OpenShiftCluster{
    Identity: &api.Identity{},
}
```